marsha/core/tests/test_api_video.py

```diff
@@ -0,0 +1,298 @@
+"""Tests for the Video API of the Marsha project."""
+import json
+
+from django.test import TestCase
+
+from rest_framework_simplejwt.tokens import AccessToken
+
+from ..factories import UserFactory, VideoFactory
+from ..models import Video
+
+
+# We don't enforce arguments documentation in tests
+# pylint: disable=missing-param-doc,missing-type-doc,unused-argument
+
+
+class VideoAPITest(TestCase):
+    """Test the API of the video object."""
+
+    def test_api_video_read_detail_anonymous(self):
+        """Anonymous users should not be allowed to read a video detail."""
+        video = VideoFactory()
+        response = self.client.get("/api/videos/{!s}/".format(video.id))
+        self.assertEqual(response.status_code, 401)
+        content = json.loads(response.content)
+        self.assertEqual(
+            content, {"detail": "Authentication credentials were not provided."}
+        )
+
+    def test_api_video_read_detail_token_user(self):
+        """A token user associated to a video should be able to read the detail of this video."""
+        video = VideoFactory()
+        playlist = video.playlist
+        jwt_token = AccessToken()
+        jwt_token.payload["video_id"] = str(video.id)
+
+        # Get the video linked to the JWT token
+        response = self.client.get(
+            "/api/videos/{!s}/".format(video.id),
+            HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
+        )
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+
+        jpeg_dict = {
+            "144": "{!s}/jpeg/{!s}_144.0000000.jpg".format(playlist.id, video.id),
```

Comment:
Maybe use the same order in `jpeg_dict` & `mp4_dict`

Reply:
done